src/mixtures/unigmm.jl

```diff
@@ -1,20 +1,22 @@
 # Univariate Gaussian Mixture Models
 
-immutable UnivariateGMM <: UnivariateMixture{Continuous,Normal}
+immutable UnivariateGMM{T<:Real} <: UnivariateMixture{Continuous,Normal}
     K::Int
-    means::Vector{Float64}
-    stds::Vector{Float64}
+    means::Vector{T}
+    stds::Vector{T}
     prior::Categorical
 
-    function UnivariateGMM(ms::Vector{Float64}, ss::Vector{Float64}, pri::Categorical)
+    (::Type{UnivariateGMM{T}}){T}(ms::Vector{T}, ss::Vector{T}, pri::Categorical) = begin
```

Comment:
Why change from the long-form `function` definition? That's generally preferable to `begin` blocks for style.